### ISSUE TITLE
BUG: Netceiver driver incompatible with parameter "-e"

### ISSUE THREAD

**Issue by lars18th** [catalinii/minisatip#230]
Hi @REELcoder ,

I found a bug in your code: When you find for "free" (!=unused) tuners, your code ins't compatible with the parameter "-e".

Imagine this environment: one internal tuner and one external Netceiver. And you like to use ONLY the external tuner. In this case you'll start minisatip with parameter "-e 1", as this DISABLES the tuner 0 (the internal tuner).

But in this case your code at:
http://github.com/catalinii/minisatip/blob/master/netceiver.c#L314
returns "na=0" and you try to register the external tuner as tuner 0. But the problem is that minisatip has assigned this value to the internal tuner, and then it is disabled. As a result, both tuners are disabled.

Perhaps you like to fix this, right? ;)
Regards!

**Comment by REELcoder** [comment]
Thanks for the hint. I see to point. But I can't fix it in netceiver.c itself. If find_netcv_adapter() is called, there is currently no way to find out if we have ZERO pointers in a[] because of disabled adapter or because none have detected. We need to add a counter or any other means, @catalinii?

**Comment by catalinii** [comment]
Hi lars,

try now.

Thanks

**Comment by REELcoder** [comment]
Hi @lars18th

Can you try with 1b578513d984e14b5b1a95d6eb5be07fe062d3b5

Thanks.

**Comment by 9000h** [comment]
there is a cosmetic issue 

```
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o minisatip.o minisatip.c 
In file included from minisatip.c:42:0:
adapter.h:119:8: warning: type defaults to ‘int’ in declaration of ‘a_count’ [-Wimplicit-int]
 extern a_count;
        ^
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o socketworks.o socketworks.c 
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o stream.o stream.c
In file included from stream.c:44:0:
adapter.h:119:8: warning: type defaults to ‘int’ in declaration of ‘a_count’ [-Wimplicit-int]
 extern a_count;
        ^
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o adapter.o adapter.c
In file included from adapter.c:33:0:
adapter.h:119:8: warning: type defaults to ‘int’ in declaration of ‘a_count’ [-Wimplicit-int]
 extern a_count;
        ^
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o utils.o utils.c
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o dvbapi.o dvbapi.c
In file included from dvbapi.h:5:0,
                 from dvbapi.c:45:
adapter.h:119:8: warning: type defaults to ‘int’ in declaration of ‘a_count’ [-Wimplicit-int]
 extern a_count;
        ^
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o ca.o ca.c
In file included from dvbapi.h:5:0,
                 from ca.c:59:
adapter.h:119:8: warning: type defaults to ‘int’ in declaration of ‘a_count’ [-Wimplicit-int]
 extern a_count;
        ^
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o tables.o tables.c
In file included from dvbapi.h:5:0,
                 from tables.c:42:
adapter.h:119:8: warning: type defaults to ‘int’ in declaration of ‘a_count’ [-Wimplicit-int]
 extern a_count;
        ^
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o satipc.o satipc.c
In file included from dvbapi.h:5:0,
                 from satipc.c:41:
adapter.h:119:8: warning: type defaults to ‘int’ in declaration of ‘a_count’ [-Wimplicit-int]
 extern a_count;
        ^
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -c -o dvb.o dvb.c
In file included from dvb.c:41:0:
adapter.h:119:8: warning: type defaults to ‘int’ in declaration of ‘a_count’ [-Wimplicit-int]
 extern a_count;
        ^
cc  -ggdb -fPIC  -DDISABLE_NETCVCLIENT -o minisatip minisatip.o socketworks.o stream.o adapter.o utils.o dvbapi.o ca.o tables.o satipc.o dvb.o -lpthread -lrt  -ldvbcsa -ldvben50221 -ldvbapi -lucsi
```

**Comment by REELcoder** [comment]
OK. fixed in 15a2f44

**Comment by REELcoder** [comment]
@lars18th: does it work for you?
